src/python/dxpy/sugar/transfers.py

```diff
@@ -0,0 +1,952 @@
+    list of local files/directories.
+
+    Args:
+        filenames (str or list of str): = Local filenames or directories.
+        remote_prefix (str) = Name to give to output tar archive. Must be provided
+            unless `filenames` is of length 1, in which case the prefix will be
+            the same as that of the single filename.
+        remote_folder (str): Optional, remote folder to upload file to.
+        compression_level (int) = Level of compression between 1 and 9 to
+            compress tar to. Specify 1 for gzip --fast and 9 for gzip --best.
+            If not specified, --fast is assumed. If None, no compression is
+            performed, i.e. the output is a .tar file.
+        chdir (bool): If a path, change to this directory; if None and filenames is
+            a directory, will cd to that directory; if True and filenames is
+            a directory, same behavior as None, otherwise an error is raised;
+            if False, no cd. (tar -C option).
+        return_handler (bool): Whether to return a DXFile handler.
+        project (str): The project ID to upload to, if not the currently selected
+            project.
+        wait_on_close (bool): Whether to block until the file has closed.
+        method (str): Compression method; one of 'gz', 'bz2'.
+        max_part_size (int): Optional, maximum file part size, defaults to project
+            value.
+
+    Returns:
+        DNAnexus link or DXFile pointing to the uploaded tar archive (depending on
+        the value of `return_handler`).
+
+    Raises:
+        ValueError: if compression_level not between 1 and 9
+        CalledProcessError: propogated from run_pipe if called command fails
+    """
+    is_dir = False
+    if isinstance(filenames, basestring):
+        is_dir = os.path.isdir(filenames)
+        filenames = [filenames]
+
+    if remote_prefix is None:
+        if len(filenames) == 1:
+            remote_prefix = os.path.basename(filenames[0])
+        else:
+            raise ValueError("'prefix' must be specified with multiple filenames")
+
+    if not remote_folder:
+        remote_folder = "/"
+    elif not remote_folder.startswith("/"):
+        remote_folder = "/{}".format(remote_folder)
+
+    if compression_level is not None and (
+        compression_level < 1 or compression_level > 9
+    ):
+        raise ValueError("Compression level must be between 1 and 9")
+
+    if compression_level:
+        zip_ext = ".{}".format(method)
+        ext = "tar" + zip_ext
+    else:
+        ext = "tar"
+
+    remote_path = "{}/{}.{}".format(remote_folder, remote_prefix, ext)
+    if project:
+        remote_path = "{}:{}".format(project, remote_path)
+
+    if not max_part_size or max_part_size <= 0:
+        max_part_size = _get_max_part_size(project)
+
+    with tmpfile() as names_file:
+        with open(names_file, "wt") as out:
+            out.write("\n".join(filenames))
```

Comment:
Replace this with:
```
with tempfile.NamedTemporaryFile(delete=False) as temp:
            with open(temp.name, 'wt') as out:
                     ...
           ...
           tar_cmd.extend(["cvf", "-", "--files-from", temp.name])
```

Reply:
But then the file doesn't get deleted after running the tar command.